### PR TITLE
Add logging about not using async for Sequential and Parallel runners

### LINKED ISSUE #2405: Inform users about not using `--async`
## Description
As discussed in https://github.com/kedro-org/kedro/issues/2036#issuecomment-1460425776 there's a possibility that users aren't aware of the `--async` feature. Instead of only logging that the async is enabled when it is, change the logging to mention that async isn't being used by default and offer steps on how to use it. 

## Context
https://github.com/kedro-org/kedro/issues/2036

## Possible Implementation
Message could be: "Asynchronous mode is disabled for loading and saving data. Use the `--async` flag for potential performance gains. https://docs.kedro.org/en/stable/nodes_and_pipelines/run_a_pipeline.html#load-and-save-asynchronously"
(Not sure if a link in logs is good practice though.. )

### PR DESCRIPTION
**Pull request by merelcht**
## Description
Closes #2405 

## Development notes
Added logging message only to `SequentialRunner` and `ParallelRunner` because async mode can't be used with `ThreadRunner`. Added tests to check the message is shown/not shown when expected.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by DimedS — COMMENTED** [review]
@merelcht , thank you for this PR,
I noticed a little bit of code repetition in 2 classes, maybe we need to consider to move this common code to AbstractRunner class, like:
```
if isinstance(self, (SequentialRunner, ParallelRunner)) and not self._is_async:
            self._logger.info(...
```

File: kedro/runner/parallel_runner.py [hunk 1]
```diff
@@ -256,6 +256,11 @@ def _run(  # noqa: too-many-locals,useless-suppression
 
         """
         # noqa: import-outside-toplevel,cyclic-import
+        if not self._is_async:
+            self._logger.info(
+                "Asynchronous mode is disabled for loading and saving data. Use the --async flag "
```
Comment (astrojuanlu) [inline kedro/runner/parallel_runner.py:261]: Somehow this reads like "We disabled asynchronous mode for loading and saving data on your behalf" (but maybe it's just me). What about

```suggestion
                "Using synchronous mode for loading and saving data. Use the --async flag "
```

**Comment by merelcht** [comment]
> @merelcht , thank you for this PR, I noticed a little bit of code repetition in 2 classes, maybe we need to consider to move this common code to AbstractRunner class, like:
> 
> ```
> if isinstance(self, (SequentialRunner, ParallelRunner)) and not self._is_async:
>             self._logger.info(...
> ```

It's a good point, but then I was thinking the `AbstractRunner` ideally shouldn't have any logic specific to the implementations, so that's why I did it on the individual classes.

**Review by DimedS — APPROVED** [review]

File: kedro/runner/parallel_runner.py (hunk 1 above)
Comment (merelcht) [reply kedro/runner/parallel_runner.py:261]: That's a fair point. I'll update the message.

File: kedro/runner/sequential_runner.py [hunk 2]
```diff
@@ -60,6 +60,11 @@ def _run(
         Raises:
             Exception: in case of any downstream node failure.
         """
+        if not self._is_async:
+            self._logger.info(
+                "Asynchronous mode is disabled for loading and saving data. Use the --async flag "
```
Comment (merelcht) [inline kedro/runner/sequential_runner.py:65]:
```suggestion
                "Using synchronous mode for loading and saving data. Use the --async flag "
```

File: tests/runner/test_parallel_runner.py [hunk 3]
```diff
@@ -73,6 +73,13 @@ def test_memory_dataset_input(self, is_async, fan_out_fan_in):
         assert len(result["Z"]) == 3
         assert result["Z"] == ("42", "42", "42")
 
+    def test_log_not_using_async(self, fan_out_fan_in, catalog, caplog):
+        catalog.add_feed_dict({"A": 42})
+        ParallelRunner().run(fan_out_fan_in, catalog)
+        assert (
+            "Asynchronous mode is disabled for loading and saving data." in caplog.text
```
Comment (merelcht) [inline tests/runner/test_parallel_runner.py:80]:
```suggestion
            "Using synchronous mode for loading and saving data." in caplog.text
```

File: tests/runner/test_sequential_runner.py [hunk 4]
```diff
@@ -29,6 +29,13 @@ def test_run_without_plugin_manager(self, fan_out_fan_in, catalog):
         assert "Z" in result
         assert result["Z"] == (42, 42, 42)
 
+    def test_log_not_using_async(self, fan_out_fan_in, catalog, caplog):
+        catalog.add_feed_dict({"A": 42})
+        SequentialRunner().run(fan_out_fan_in, catalog)
+        assert (
+            "Asynchronous mode is disabled for loading and saving data." in caplog.text
```
Comment (merelcht) [inline tests/runner/test_sequential_runner.py:36]:
```suggestion
            "Using synchronous mode for loading and saving data." in caplog.text
```

File: tests/runner/test_thread_runner.py [hunk 5]
```diff
@@ -34,6 +34,14 @@ def test_memory_dataset_input(self, fan_out_fan_in):
         assert "Z" in result
         assert result["Z"] == ("42", "42", "42")
 
+    def test_does_not_log_not_using_async(self, fan_out_fan_in, catalog, caplog):
+        catalog.add_feed_dict({"A": 42})
+        ThreadRunner().run(fan_out_fan_in, catalog)
+        assert (
+            "Asynchronous mode is disabled for loading and saving data."
```
Comment (merelcht) [inline tests/runner/test_thread_runner.py:41]:
```suggestion
            "Using synchronous mode for loading and saving data."
```

**Review by astrojuanlu — APPROVED** [review]

**Pull request merged**